numpyro/mcmc.py

```diff
@@ -53,24 +55,32 @@ def init_kernel(init_samples,
                     diag_mass=True,
                     target_accept_prob=0.8,
                     trajectory_length=2*math.pi,
+                    max_tree_depth=10,
                     run_warmup=True,
+                    use_prims=True,
+                    heuristic_step_size=True,
                     rng=PRNGKey(0)):
         step_size = float(step_size)
-        nonlocal momentum_generator, wa_update, trajectory_len
+        nonlocal momentum_generator, wa_update, trajectory_len, max_treedepth
         trajectory_len = float(trajectory_length)
+        max_treedepth = max_tree_depth
         z = init_samples
         z_flat, unravel_fn = ravel_pytree(z)
         momentum_generator = partial(_sample_momentum, unravel_fn)
 
-        find_reasonable_ss = partial(find_reasonable_step_size,
-                                     potential_fn, kinetic_fn, momentum_generator)
+        wa_kwargs = {}
+        # FIXME: compiling find_reasonable_step_size is so slow for some models
+        if heuristic_step_size:
```

Comment:
Can you elaborate why we need this, and setting `adapt_step_size=False` is not sufficient?

Reply:
With this, we still adapt step_size but do not tuning step_size when mass_matrix is updated (at the end of each window). Indeed, tuning step_size is not important under dual averaging scheme.

Reply:
More context about this: there is a bug in Stan (even with latest version), where `find_reasonable_step_size` does not run at beginning of warmup phase; but step_size will change very fast based on `accept_prob` value. If given a step_size, accept_prob is small, then step_size will reduce a lot under dual averaging scheme.

However, tuning step_size is useful to help NUTS/HMC not spend so many steps at initial steps of each window. For example, if `inv_mm` changes from `1 -> 0.001` then `find_reasonable_ss` will help increase step_size to a much larger value than the current value.

Reply:
> With this, we still adapt step_size but do not tuning step_size when mass_matrix is updated (at the end of each window).

IIRC, we weren't doing this in Pyro initially either, but found that updating step size when mass matrix is updated helps prevent cases where we make very slow progress due to the mismatch that you mentioned above. So, in what situations, do you think turning this off is advisable? Is this mostly needed because compilation of `find_reasonable_step_size` is slow on larger models?

Reply:
Yes, it is only there until this `FIXME` is resolved.